python/tests/io/test_fsspec.py

```diff
@@ -0,0 +1,186 @@
+        f.write(b"foobar")
+
+    input_file = fsspec_fileio.new_input(location=f"s3://warehouse/{filename}")
+    f = input_file.open()
+
+    f.seek(0)
+    assert f.tell() == 0
+    f.seek(1)
+    assert f.tell() == 1
+    f.seek(3)
+    assert f.tell() == 3
+    f.seek(0)
+    assert f.tell() == 0
+
+
+@pytest.mark.s3
+def test_fsspec_read_specified_bytes_for_file(fsspec_fileio):
+    """Test reading a specified number of bytes from an fsspec file-io file"""
+
+    filename = str(uuid.uuid4())
+    output_file = fsspec_fileio.new_output(location=f"s3://warehouse/{filename}")
+    with output_file.create() as f:
+        f.write(b"foo")
+
+    input_file = fsspec_fileio.new_input(location=f"s3://warehouse/{filename}")
+    f = input_file.open()
+
+    f.seek(0)
+    assert b"f" == f.read(1)
+    f.seek(0)
+    assert b"fo" == f.read(2)
+    f.seek(1)
+    assert b"o" == f.read(1)
+    f.seek(1)
+    assert b"oo" == f.read(2)
+    f.seek(0)
+    assert b"foo" == f.read(999)  # test reading amount larger than entire content length
+
+    fsspec_fileio.delete(input_file)
+
+
+@pytest.mark.s3
+def test_fsspec_raise_on_opening_file_not_found(fsspec_fileio):
+    """Test that an fsppec input file raises appropriately when the s3 file is not found"""
+
+    filename = str(uuid.uuid4())
+    input_file = fsspec_fileio.new_input(location=f"s3://warehouse/{filename}")
+    with pytest.raises(FileNotFoundError) as exc_info:
+        input_file.open().read()
+
+    assert filename in str(exc_info.value)
+
+
+@pytest.mark.s3
+def test_checking_if_a_file_exists(fsspec_fileio):
+    """Test checking if a file exists"""
+
+    non_existent_file = fsspec_fileio.new_input(location="s3://warehouse/does-not-exist.txt")
+    assert not non_existent_file.exists()
+
+    filename = str(uuid.uuid4())
+    output_file = fsspec_fileio.new_output(location=f"s3://warehouse/{filename}")
+    assert not output_file.exists()
+    with output_file.create() as f:
+        f.write(b"foo")
+
+    existing_input_file = fsspec_fileio.new_input(location=f"s3://warehouse/{filename}")
+    assert existing_input_file.exists()
+
+    existing_output_file = fsspec_fileio.new_output(location=f"s3://warehouse/{filename}")
+    assert existing_output_file.exists()
+
+    fsspec_fileio.delete(existing_output_file)
+
+
+@pytest.mark.s3
+def test_closing_a_file(fsspec_fileio):
+    """Test closing an output file and input file"""
+    filename = str(uuid.uuid4())
+    output_file = fsspec_fileio.new_output(location=f"s3://warehouse/{filename}")
+    with output_file.create() as f:
+        f.write(b"foo")
+        assert not f.closed
+    assert f.closed
+
+    input_file = fsspec_fileio.new_input(location=f"s3://warehouse/{filename}")
+    f = input_file.open()
+    assert not f.closed
+    f.close()
+    assert f.closed
+
+    fsspec_fileio.delete(f"s3://warehouse/{filename}")
+
+
+@pytest.mark.s3
+def test_fsspec_converting_an_outputfile_to_an_inputfile(fsspec_fileio):
```

Comment:
Can we have a test that writes an Avro file and reads it back? It would be good to have a bit more real world test cases.

Reply:
Good idea, I added a test called `test_writing_avro_file` that reads a local avro file, writes it to a location with an `s3` scheme using fsspec-file-io, then reads it back and asserts that the bytes match.